docs/pages/versions/unversioned/sdk/battery.mdx

```diff
@@ -23,44 +23,41 @@ import { SnackInline} from '~/ui/components/Snippet';
 <SnackInline label='Basic Battery Usage' dependencies={['expo-battery']}>
 
 ```jsx
-import * as React from 'react';
-import * as Battery from 'expo-battery';
-import { StyleSheet, Text, View } from 'react-native';
+import React, { useEffect, useState } from "react";
+import * as Battery from "expo-battery";
+import { StyleSheet, Text, View } from "react-native";
+
+export default function App() {
+  const [batteryLevel, setBatteryLevel] = useState(null);
+  const [subscription, setSubscription] = useState(null);
 
-export default class App extends React.Component {
-  state = {
-    batteryLevel: null,
+  const _subscribe = async () => {
+    const batteryLevel = await Battery.getBatteryLevelAsync();
+    setBatteryLevel(batteryLevel);
+    
+    setSubscription(
+      Battery.addBatteryLevelListener(({ batteryLevel }) => {
+        setBatteryLevel(batteryLevel);
+        console.log("batteryLevel changed!", batteryLevel);
+      })
+    );
   };
 
-  componentDidMount() {
-    this._subscribe();
-  }
+  const _unsubscribe = () => {
+    subscription && subscription.remove();
+    setSubscription(null);
+  };
 
-  componentWillUnmount() {
-    this._unsubscribe();
-  }
+  useEffect(() => {
+    _subscribe();
+    return () => _unsubscribe();
+  }, []);
```

Comment:
Let's remove `uesEffect` dependency array, or make sure that there are correct values specified, so there is no warning, when opening example in Snack.

Reply:
Decided to fix warnings as suggested by the Snack editor.